### PR TITLE
Upgrade testing for Wagtail 6.3

### PR DESCRIPTION
**Pull request by nickmoreton**
This pull request includes: updates to the CI workflows and testing configurations to remove support for Python 3.8 and Django 3.2, and to add support for newer versions of Django and Wagtail.

### CI Workflow Updates:
* [`.github/workflows/ci.yml`](diffhunk://#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL30-R30): Removed Python 3.8 from the matrix of Python versions.
* [`.github/workflows/ruff.yml`](diffhunk://#diff-2648f2b57520d4056946a7e509a50cfed47a7a8b389c0861011416ba78ae0db5L17-R26): Updated the Ruff workflow to use `actions/setup-python@v5` and install dependencies with Python 3.12.

### Dependency Management:
* [`pyproject.toml`](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L17-R24): Removed classifiers for Python 3.8 and Django 3.2, and added a classifier for Django 5.1.

### Testing Configuration:
* [`tox.ini`](diffhunk://#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L5-L11): Updated the `envlist` to remove Python 3.8 and add configurations for Django 5.1 and newer versions of Wagtail.
* [`tox.ini`](diffhunk://#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L31-R36): Updated `deps` to remove Django 3.2 and add Django 5.1 and newer versions of Wagtail.

### CODE REVIEW

**Comment by nickmoreton** [comment]
There is a deprecation warning for: https://docs.wagtail.org/en/stable/releases/6.0.html#deprecated-widgetwithscript-base-widget-class but I think that should be a separate piece of work and isn't included here.

File: .github/workflows/ruff.yml [hunk 1]
```diff
@@ -13,10 +13,14 @@ jobs:
...
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff
+      - name: Run Ruff
+        run: ruff check --output-format=github ./src/wagtailmarkdown
```
Comment (zerolab) [inline .github/workflows/ruff.yml:26]: These changes are not needed.

What we do need is to install the same ruff version as in pre-commit https://github.com/torchbox/wagtail-markdown/blob/main/.pre-commit-config.yaml#L23C11-L23C17

a la https://github.com/torchbox/wagtail-periodic-review/blob/main/.github/workflows/ruff.yml

File: tox.ini [hunk 2]
```diff
@@ -2,13 +2,12 @@
 min_version = 4.0
 
 envlist =
-    py{38,39,310}-django3.2-wagtail5.2
-    py{38,39,310,311}-django4.2-wagtail{5.2,6.0}
-    py{311,312}-django5.0-wagtail{5.2,6.0,main}
+    py{39,310,311}-django4.2-wagtail{5.2,6.2,6.3}
+    py{311,312}-django5.0-wagtail{5.2,6.2,6.3}
+    py{311,312}-django5.1-wagtail{6.3,main}
```
Comment (zerolab) [inline tox.ini:7]: let's add Python 3.13 to the mix too

**Comment by zerolab** [comment]
closing in favour of #149

**Pull request closed without merging**